src/AccessibilityInsights.SetupLibrary/ExecutionHistory.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+
+namespace AccessibilityInsights.SetupLibrary
+{
+    /// <summary>
+    /// Class to capture the exeuction history of the VersionSwitcher. Gets serialized to disk by
+    /// the VersionSwitcher, then gets uploaded to telemetry the next time the app boots
+    /// </summary>
+    public class ExecutionHistory
+    {
+        public static string GetDataFilePath()
+        {
+            const string TempVariable = "temp";
+            const string DataFile = "AccessibilityInsights.VersionSwitcher.ExecutionHistory.json";
+            return Path.Combine(Environment.GetEnvironmentVariable(TempVariable), DataFile);
+        }
+
+        private ExecutionResult _typedExecutionResult;
+
+        public ExecutionHistory()
+        {
+            LocalDetails = new List<string>();
+
+            // This forces explicit initialization of VersionSwitcherResult
```

Comment:
This comment is confusing to me- can you expand?

Reply:
Reworked the comment, including fixing the name mismatch from when ExecutionResult was called VersionSwitcherResult